### PR TITLE
Feature/seext 12795 create animated horizontal scrolling text and skip track svg icon

### PR DESCRIPTION
**Pull request by tomislav-arambasic**
## Summary
- Added `AnimatedScrollText`
  - If text is wider than container it's wrapped in, text will be scrolled to the end, then back to start, then wait for X seconds before starting that animation again
  - Animation speed is configurable, with defined default config
- Added `skip-next` and `playlist-play` icons

**Icons**
Screenshot 2024-03-15 at 16 33 38
Screenshot 2024-03-15 at 16 33 33

**Animation videos**

https://github.com/shoutem/ui/assets/57353746/a195216d-b9fd-43b2-8a50-e1f7287aea77

https://github.com/shoutem/ui/assets/57353746/5dfa1423-48d2-4a12-8542-c4eaba510511

### CODE REVIEW

File: components/SvgIcons/SkipTrackIcon.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  return (
+    <View style={container}>
+      <Svg width={size} height={size} viewBox="0 0 24 24">
+        <Path
+          d={`M0 ${12 -
+            halfTriangleHeight} l${triangleWidth} ${halfTriangleHeight} l-${triangleWidth} ${halfTriangleHeight} Z`}
+          fill={color}
+        />
+        <Path
+          d={`M${triangleWidth} ${12 - halfTriangleHeight} v${triangleHeight}`}
```
Comment (Definitely-Not-Vlad) [inline components/SvgIcons/SkipTrackIcon.js:29]:
```suggestion
  const halfTriangleHeight = triangleHeight / 2;
  const pathOne = `M0 ${12 - halfTriangleHeight} l${triangleWidth} ${halfTriangleHeight} l-${triangleWidth} ${halfTriangleHeight} Z`;
  const pathTwo = `M${triangleWidth} ${12 - halfTriangleHeight} v${triangleHeight}`;
  
  return (
    <View style={container}>
      <Svg width={size} height={size} viewBox="0 0 24 24">
        <Path
          d={pathOne}
          fill={color}
        />
        <Path
          d={pathTwo}
```

Just a suggestion add it if you like it (:

**Review by Definitely-Not-Vlad — APPROVED** [review]
LGTM, just added a small suggestion, not a must, up to you!

**Review by sstimac — APPROVED** [review]
Looks good. Only a couple of Qs

File: components/AnimatedScrollingText.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  const handleContainerLayout = e => {
+    setContainerWidth(e.nativeEvent.layout.width);
+  };
+
+  const handleTextLayout = e => {
+    setTextWidth(e.nativeEvent.layout.width);
+  };
+
+  return (
+    <ScrollView
```
[the remark this answers is not in the document]
Comment (tomislav-arambasic) [reply components/AnimatedScrollingText.js:64]: yep

File: components/SvgIcons/SkipTrackIcon.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { View } from 'react-native';
+import Svg, { Path } from 'react-native-svg';
+import PropTypes from 'prop-types';
+import { connectStyle } from '@shoutem/theme';
+
+const SkipTrackIcon = ({ style }) => {
```
[the remark this answers is not in the document]
Comment (tomislav-arambasic) [reply components/SvgIcons/SkipTrackIcon.js:7]: Why would we want to make any icon generic?
The point of this - we don't have designer to create our icons any more. We're creating them programatically.
Why would we ever want anything about icons to be generic?
Even if we do, I wouldn't want to overcomplicate things, it's an icon, nothing else.

[the remark this answers is not in the document]
Comment (sstimac) [reply components/SvgIcons/SkipTrackIcon.js:7]: My point is now we have Icons and SvgIcons in the repo. Why not go for the approach where you can use this SVG via our existing Icon component ? It accepts the most style properties via props, except for the precise stroke widths

**Pull request merged**